app/Auth/Access/RegistrationService.php

```diff
@@ -72,6 +73,13 @@ public function findOrRegister(string $name, string $email, string $externalId):
             ];
 
             $user = $this->registerUser($userData, null, false);
+        } else if ($updateUser && ($user->name !== $name || $user->email !== $email)) {
+            $user->update([
+                'name'  => $name,
+                'email' => $email,
+            ]);
+
```

Comment:
Could this logic be moved to it's own method within this class instead? 
Passing a flag argument to trigger extra logic starts to make this method a little muddy, would prefer a focused method that's optionally called (depending on setting) from the Saml2service side.
If you want to specifically only update new instances I think there might be something within the Eloquent model for that, or you could resort to checking the created date.